web-registry/src/hooks/useMsgClient.ts

```diff
@@ -0,0 +1,68 @@
+import { useState, useCallback } from 'react';
+import { StdFee, DeliverTxResponse } from '@cosmjs/stargate';
+
+import { useLedger } from '../ledger';
+import { Sender } from '../lib/wallet';
+
+type MsgClientType = {
+  signAndBroadcast: (msgs: any, fee?: StdFee, memo?: string) => Promise<void>;
+  setDeliverTxResponse: (txResult: DeliverTxResponse | undefined) => void;
+  deliverTxResponse?: DeliverTxResponse;
+  error?: unknown;
+  wallet?: Sender;
+};
+
+export default function useMsgClient(
+  handleTxQueued: () => void,
```

Comment:
You can optionally pass in the raw TxBytes here. I think my plan was to get the in-progress tx hash from this somehow, so we can display it in the Processing Modal. However, I'm not sure it is possible. Just a heads up. No need to change yet.

```suggestion
  handleTxQueued: (txBytes: Uint8Array) => void,
```

Reply:
we could add that if we want to display the tx hash in the processing modal but this is not the current design